http-webhook.md

```diff
@@ -0,0 +1,339 @@
+# HTTP 1.1 Web Hooks for Event Delivery
+
+## Abstract
+
+"Web hooks" are a popular pattern to deliver notifications between applications
+and via HTTP endpoints. In spite of pattern usage being widespread, there is no
+formal definition for Web Hooks. This specification aims to provide such a
+definition for use with [CNCF CloudEvents][CE], but is considered generally
+usable beyond the scope of CloudEvents.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+- 1.1. [Conformance](#11-conformance)
+- 1.2. [Relation to HTTP](#12-relation-to-http)
+2. [Delivering notifications](#2-delivering-notifications)
+3. [Authorization](#3-authorization)
+4. [Abuse Protection](#4-abuse-protection)
+5. [References](#5-references)
+
+## 1. Introduction
+
+"Web hooks" are a popular pattern to deliver notifications between applications
+and via HTTP endpoints. Applications that make notifications available, allow
+for other applications to register an HTTP endpoint to which notifications are
+delivered once available.
+
+This specification defines a HTTP method by how notifications are delivered by
+the sender, an authorization model for event delivery to protect the receiver,
+and a registration handshake that protects the sender from being abused for
+flooding arbitrary HTTP sites with requests.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][RFC2119].
+
+### 1.2. Relation to HTTP
+
+This specification prescribes rules constraining the use and handling of
+specific [HTTP methods][RFC7231-Section-4] and headers.
+
+This specification also applies equivalently to HTTP/2 ([RFC7540][RFC7540]),
+which is compatible with HTTP 1.1 semantics.
+
+## 2. Delivering notifications
+
+### 2.1. Delivery request
+
+Notifications are delivered using a HTTP request. The response indicates the
+resulting status of the delivery.
+
+HTTP-over-TLS (HTTPS) [RFC2818][RFC2818] MUST be used for the connection.
+
+The HTTP method for the delivery request MUST be [POST][POST].
+
+The [`Content-Type`][Content-Type] header MUST be carried and the request MUST
```

Comment:
What if data is specified via the query string and there is no payload? Are you saying every webhook MUST send a payload? If so, why?

Reply:
@glennblock you've pointed to the Content-Length rule yourself, Glenn. POST requires a payload body, but that body may be empty and even if it's empty you probably still need to declare it. That said, in the interest of trading choice for interop, I do indeed believe that a payload body can be mandated for a webhook request.